### PR TITLE
Blog post about a factoring magic trick

### PR DESCRIPTION
**Pull request by RobinJadoul**
I'm not sure of what tags would be appropriate, so those are blank for now.
Perhaps we should also make the original author more anonymous than they currently are

### CODE REVIEW

File: _posts/2021-01-22-logarithmic-time-factoring-magic-trick.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+layout: post
+title: "Factoring in logarithmic time: a mathemagic trick"
+categories:
+author: Robin Jadoul
+meta: "Factoring in logarithmic time: a mathemagic trick"
+tags:
+excerpt_separator: <!--more-->
+---
+
+When you have an interest in cryptography, and you occasionally browse reddit, you've probably seen something like this before. Someone creates a new post on r/crypto or r/cryptography, claiming to be able to efficiently factor integers or RSA moduli. This then usually goes hand in hand with a serious lack of context and explanation, notation that is questionable at best, often just plain unreadable, and an aversion of providing proper proof by factoring an actual, cryptographically big modulus. Yet because it's hard to understand, and somethings there's math involved that at first glance seems like it might have some deeper meaning, you're left wondering, is there some insight to be found here.
```
Comment (hyperreality) [inline _posts/2021-01-22-logarithmic-time-factoring-magic-trick.md]: of providing proper proof -> to providing minimal sufficient proof (too many p's?)
somethings -> sometimes

File: _posts/2021-01-22-logarithmic-time-factoring-magic-trick.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
+---
+layout: post
+title: "Factoring in logarithmic time: a mathemagic trick"
+categories:
+author: Robin Jadoul
+meta: "Factoring in logarithmic time: a mathemagic trick"
+tags:
```
Comment (hyperreality) [inline _posts/2021-01-22-logarithmic-time-factoring-magic-trick.md]: tags: RSA Broken-Cryptography?

File: _posts/2021-01-22-logarithmic-time-factoring-magic-trick.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+![Image containing mostly unreadable math with a small RSA modulus factored by hand as example](https://cdn.discordapp.com/attachments/695239931790098434/801892927587876864/Screenshot_2021-01-21_at_7.16.05_pm.png)
+
+The claim in the title is immediately very interesting: "Integer factorization in log[(N + 1)/2]/log[2]". We'll come back to the notation in a bit, but we can see at a glance that the author claims to solve factorization in logarithmic time (so in polynomial/linear time in the number of bits, commonly known as "efficient"). Anyone with a passing knowledge on the topic of cryptography, and more in particular of RSA, will know that this is thought to be a hard problem. So either someone just made an enormous breakthrough, and we need to start changing a lot of the underlying infrastructure of most of our previously thought to be secure communication, or (and this is the far more likely option), there's something wrong with the proof.
```
Comment (hyperreality) [inline _posts/2021-01-22-logarithmic-time-factoring-magic-trick.md]: previously thought to be secure -> previously thought-to-be-secure ?

File: _posts/2021-01-22-logarithmic-time-factoring-magic-trick.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+### Line 1: perhaps it doesn't work for general N
+
+On the very first line, the integer being used as input to the described algorithm, $N$ is said to be of the form $pq$, with p and q prime. So far so good.
+
+Where it gets a bit weird however, is that it is said that $p = 4a + 3$ and $q = 4a + 1$. This knowledge would of course immediately enable us to factor $N$ from the quadratic equation arising from the twin prime substitution $q = p + 2$. Let's assume this is just some unfortunate notation, and what's in fact meant is that $p \equiv 3 \pmod 4$ and $q \equiv 1 \pmod 4$. Reading this, we know we might want to keep an eye out if this assumption appears anywhere later on, or if it's even more general, and all that's needed is that $p$ and $q$ are simply odd primes.
+
+### Line 2: a wild formula appeared
+
+The mathematical notation is not very great, so let's first try to make it a bit more readable. We quickly spot a `log[...]/log[2]`, which of course is simply $\log_2(\ldots)$. From there, most of it is fairly easily transcribed and rendered with the ever-magical $\LaTeX$:
```
Comment (hyperreality) [inline _posts/2021-01-22-logarithmic-time-factoring-magic-trick.md]: Remove "of course"

File: _posts/2021-01-22-logarithmic-time-factoring-magic-trick.md (hunk 2 above)
Comment (RobinJadoul) [reply _posts/2021-01-22-logarithmic-time-factoring-magic-trick.md]: Is there a difference between tags and categories? How should those be treated?

Comment (hyperreality) [reply _posts/2021-01-22-logarithmic-time-factoring-magic-trick.md]: I don't think Category is being used in any way yet. Jack has some ideas about how we could use it in future, but you could just put "News" here for now as a suitably generic category?

File: _posts/2021-01-22-logarithmic-time-factoring-magic-trick.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+**Spoiler alert**: nope...
+<!--more-->
+
+## Finding the good stuff
+
+As a case study, we'll have a closer look at [this reddit post](https://www.reddit.com/r/cryptography/comments/l1zf2y/is_this_known_as_an_attack_on_rsa_what_are_the/), which asks the question if this is a known attack, and then enigmatically just links to a post on [linkedin](https://www.linkedin.com/feed/update/urn:li:ugcPost:6757583973079117824/). To get around my lack of linkedin account, someone was kind enough to send me the following image of the main content.
+
+![Image containing mostly unreadable math with a small RSA modulus factored by hand as example](https://cdn.discordapp.com/attachments/695239931790098434/801892927587876864/Screenshot_2021-01-21_at_7.16.05_pm.png)
```
Comment (hyperreality) [inline _posts/2021-01-22-logarithmic-time-factoring-magic-trick.md]: As for anonymity, can you perhaps remove the linkedin link (will probably go invalid anyway), and remove the contact details from the bottom of the image - then put it in assets/images and you can refer to it as ``

Comment (RobinJadoul) [reply _posts/2021-01-22-logarithmic-time-factoring-magic-trick.md]: Hmm, tried to amend + force push, but it's still here in the discussion history

Comment (RobinJadoul) [reply _posts/2021-01-22-logarithmic-time-factoring-magic-trick.md]: The commit still lingers somewhere on github, and I don't know if there's even a way to get it gc'd, but at least few people would come looking here, for this, anyway

**Pull request merged**